browser/ai_chat/ai_chat_render_view_context_menu_browsertest.cc

```diff
@@ -154,15 +122,42 @@ class AIChatRenderViewContextMenuBrowserTest : public InProcessBrowserTest {
                 base::StringPrintf("getRectY('%s')", element_id.c_str()))
                 .ExtractInt();
 
-    // Calls ConversationDriver::SubmitSelectedText
-    ContextMenuNotificationObserver context_menu_observer(
-        IDC_AI_CHAT_CONTEXT_SHORTEN);
+    RenderViewContextMenu::RegisterMenuShownCallbackForTesting(
+        base::BindLambdaForTesting([&](RenderViewContextMenu* context_menu) {
+          auto* brave_context_menu =
+              static_cast<BraveRenderViewContextMenu*>(context_menu);
+          brave_context_menu->SetAIEngineForTesting(
+              std::make_unique<MockEngineConsumer>());
+          ai_engine = static_cast<MockEngineConsumer*>(
+              brave_context_menu->GetAIEngineForTesting());
+          // Verify that rewrite is requested
+          EXPECT_CALL(*ai_engine, GenerateRewriteSuggestion(_, _, _, _))
+              .WillOnce(
+                  [&](std::string text, const std::string& question,
+                      EngineConsumer::GenerationDataCallback data_callback,
+                      EngineConsumer::GenerationCompletedCallback callback) {
+                    ASSERT_TRUE(callback);
+                    ASSERT_TRUE(data_callback);
+                    for (const auto& data : received_data) {
+                      auto event =
+                          mojom::ConversationEntryEvent::NewCompletionEvent(
+                              mojom::CompletionEvent::New(data));
+                      data_callback.Run(std::move(event));
+                    }
+                    std::move(callback).Run(completed_result);
+                    run_loop.Quit();
+                  });
+          base::SingleThreadTaskRunner::GetCurrentDefault()->PostTask(
+              FROM_HERE, base::BindOnce(&ExecuteRewriteCommand, context_menu));
```

Comment:
Why do we need to post executing rewrite command here?

Reply:
This is what the previous code did with `ContextMenuNotificationObserver` https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/renderer_context_menu/render_view_context_menu_browsertest_util.cc;l=37-40?q=ContextMenuNotificationObserver

Since that helper doesn't allow an additional intercepter for getting a handle to the `RenderViewContextMenu`, I had to replicate the basic functionality it does so that I could replace the EngineConsumer. I assume it has a good reason for posting to thread.